frontend/__tests__/storylines.js

```diff
@@ -540,36 +561,175 @@ describe("Storylines", () => {
+        expect(stubStoryline.testPropositionalCondition({
+          _type: "propositional_condition",
+          boolean_operator: "OR",
+          conditions: [
+            {
+              _type: "atomic_condition",
+              lhs: buildState(['global', 'foo']),
+              operator: "==",
+              rhs: "bar"
+            },
+            {
+              _type: "atomic_condition",
+              lhs: true,
+              operator: "==",
+              rhs: true
+            }
+        ]})).toBeTruthy();
+      });
+
+      it("should return false when using OR and all conditions fails", () => {
+        stubStoryline.state.global = getFooEqualBarState();
+        expect(stubStoryline.testPropositionalCondition({
+          _type: "propositional_condition",
+          boolean_operator: "OR",
+          conditions: [
+            {
+              _type: "atomic_condition",
+              lhs: buildState(['global', 'foo']),
+              operator: "==",
+              rhs: "nope"
+            },
+            {
+              _type: "atomic_condition",
+              lhs: true,
+              operator: "==",
+              rhs: false
+            }
+        ]})).toBeFalsy();
+      });
+
+      it("should fail on invalid boolean operator", () => {
+        stubStoryline.state.global = getFooEqualBarState();
+        expect(() => stubStoryline.testPropositionalCondition({
+          _type: "propositional_condition",
+          boolean_operator: "XOR",
+          conditions: []})).toThrow(/Invalid boolean operator/i);
+      });
+    });
+
+    describe("testCondition()", () => {
+      it("should recursively evaluate conditions (to true)", () => {
+        stubStoryline.state.global = getFooEqualBarState();
+        expect(stubStoryline.testCondition({
+          _type: "propositional_condition",
+          boolean_operator: "AND",
+          conditions: [
+            {
+              _type: "atomic_condition",
+              lhs: buildState(['global', 'foo']),
+              operator: "==",
+              rhs: "bar"
+            },
+            {
+              _type: "propositional_condition",
+              boolean_operator: "OR",
+              conditions: [{
+                _type: "atomic_condition",
+                lhs: true,
+                operator: "==",
+                rhs: true
+              }]
+            }
+        ]})).toBeTruthy();
+      });
+
+      it("should recursively evaluate conditions (to false)", () => {
+        stubStoryline.state.global = getFooEqualBarState();
+        expect(stubStoryline.testCondition({
+          _type: "propositional_condition",
+          boolean_operator: "AND",
+          conditions: [
+            {
+              _type: "atomic_condition",
+              lhs: buildState(['global', 'foo']),
+              operator: "==",
+              rhs: "bar"
+            },
+            {
+              _type: "propositional_condition",
+              boolean_operator: "OR",
```

Comment:
This is testing lazy evaluation, not just recursive evaluation, because there is only one condition in this 'OR'. Because this one is different, I'm guessing you meant to test lazy evaluation?

Reply:
Nope, but I can add one test for lazy evaluation.

Reply:
So I'm not going to test lazy evaluation as that's an implementation details, but I'll test what happend on conditions.length === 1 and condition.length > 2